### PR TITLE
Null handling when there is no data source in LocalConnectionReferences

### PR DESCRIPTION
**Pull request by suskumar-MSFT**

### CODE REVIEW

File: src/PAModel/CanvasDocument.cs [hunk 1]
```diff
@@ -492,13 +492,16 @@ internal void OnLoadComplete(ErrorContainer errors)
                     errors.FormatNotSupported($"Document consistency error. Connection id mismatch");
                     throw new DocumentException();
                 }
-                foreach (var dataSourceName in connection.dataSources)
+                if (connection.dataSources != null)
                 {
-                    var ds = _dataSources.SelectMany(x => x.Value).Where(x => x.Name == dataSourceName).FirstOrDefault();
-                    if (ds == null)
+                    foreach (var dataSourceName in connection.dataSources)
```
Comment (lesaltzm) [inline src/PAModel/CanvasDocument.cs:497]: I generally find writing this as 
`foreach (var dataSourceName in connection.dataSources ?? Enumerable.Empty())` 
to be a bit cleaner, but it depends on if you want to emphasize the null check or not.

**Review by lesaltzm — APPROVED** [review]

File: src/PAModel/CanvasDocument.cs (hunk 1 above)
Comment (MikeStall) [reply src/PAModel/CanvasDocument.cs:497]: We have an extension method (NullOk<T> ) for this too, you could writer...  

foreach (var dataSourceName in connection.dataSources.NullOk())

(see line 486). Turns out extension methods can accept null :D

**Review by MikeStall — APPROVED** [review]
:shipit:

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**